Arithmetic tests

jenswi-linaro opened the pull request.

etienne-lms commented on host/xtest/regression_4100.c:
```diff
@@ -0,0 +1,2536 @@
+		"Test TEE Internal API Arithmetical API - Neg");
+
+static bool do_mul(ADBG_Case_t *c, TEEC_Session *s, const char *str_s,
+		      const char *str_t, const char *str_r)
```
few minor indentation issues: some manual space indent that do not match.
Check also definition header for `test_4110_mod()` and `test_4110_fmm()`.

jenswi-linaro commented:
Addressed all comments but the one about "CHECK_PT()"

jenswi-linaro commented:
Update

jforissier commented:
@etienne-lms ping?

etienne-lms commented on host/xtest/regression_4100.c:
```diff
@@ -0,0 +1,2539 @@
+
+	res = true;
+out:
+	if (ha != TA_CRYPT_ARITH_INVALID_HANDLE)
```
minor: condition could be removed.

jenswi-linaro replied:
I'll fix all threse

etienne-lms commented on host/xtest/regression_4100.c:
```diff
@@ -0,0 +1,2539 @@
+
+	/* First do the ADD dominated phase */
+
+
```
minor: 2 empty lines

jenswi-linaro replied:
I'll fix all these

etienne-lms commented on host/xtest/regression_4100.c:
```diff
@@ -0,0 +1,2539 @@
+
+	res = true;
+out:
+	if (hn != TA_CRYPT_ARITH_INVALID_HANDLE)
```
minor: condition could be removed.

Few other occurrences below.

etienne-lms commented:
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`
Minor things that could be addressed. Very clean test integration.

jenswi-linaro commented:
Comments addressed, tags applied, squashed and rebased.
I've restarted the build of https://github.com/OP-TEE/optee_os/pull/2682 to see that everything builds and tests OK

jenswi-linaro commented:
Tested OK on Hikey, added a tag for that too.
